### PR TITLE
Issue #493: When GitHub token is invalid/expired, provide information which token provider was used

### PR DESCRIPTION
**Pull request by amalota**

### CODE REVIEW

File: git_machete/github.py [hunk 1]
```diff
@@ -131,6 +139,14 @@ def get_token_from_file_in_home_directory() -> Optional[str]:
             get_token_from_hub())
 
 
+def __get_github_token() -> Optional[str]:
+    return __get_github_token_and_provider().token if __get_github_token_and_provider() else None
```
Comment (PawelLipski) [inline git_machete/github.py]: Nit: I'd extract the result of `__get_github_token_and_provider()` to a variable to avoid a double call

**Comment by amalota** [comment]
Example messages:
Screenshot 2022-06-10 at 19 20 41

image

@PawelLipski I was thinking i could also add info that you can change the way you provide token and display possible providers as before, wdyt?

**Comment by PawelLipski** [comment]
> i could also add info that you can change the way you provide token and display possible providers as before, wdyt?

Not bad an idea.. in case token is already present, you can just mention that the user can check different providers in the help, without listing them explicitly (as too long error messages cause TL;DR reactions inevitably)

File: git_machete/github.py [hunk 2]
```diff
@@ -59,9 +59,13 @@ def __parse_pr_json(pr_json: Any) -> GitHubPullRequest:
                              repository_url=pr_json['head']['repo']['html_url'] if pr_json['head']['repo'] else None)
 
 
-def __get_github_token() -> Optional[str]:
-    def get_token_from_gh() -> Optional[str]:
+class GithubTokenAndTokenProvider(NamedTuple):
```
Comment (PawelLipski) [inline git_machete/github.py:62]: Pls bump version + mention this entire change in release notes

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**